server/app/models.py

```diff
@@ -1108,6 +1103,11 @@ def backup(self):
 
     @classmethod
     def _can(cls, user, need, final, query):
```

Comment:
We should say that you can't touch the final submission at all after the lock date unless you're an admin.